src/analysis/bounded_variation.lean

```diff
@@ -609,47 +610,84 @@ begin
   simpa only [tsub_pos_iff_lt, finset.mem_range] using hx,
 end
 
-lemma comp_eq_of_monotone_on (f : α → E) {s : set α} {t : set β} (φ : β → α)
-  (hφ : monotone_on φ t) (φst : set.maps_to φ t s) (φsur : set.surj_on φ t s) :
-  evariation_on (f ∘ φ) t = evariation_on f s :=
+lemma comp_eq_of_monotone_on (f : α → E) {t : set β} (φ : β → α) (hφ : monotone_on φ t) :
+  evariation_on (f ∘ φ) t = evariation_on f (φ '' t) :=
 begin
-  apply le_antisymm (comp_le_of_monotone_on f φ hφ φst),
+  apply le_antisymm (comp_le_of_monotone_on f φ hφ (t.maps_to_image φ)),
   casesI is_empty_or_nonempty β,
   { convert zero_le _,
-    exact evariation_on.subsingleton f ((subsingleton_of_subsingleton.image _).anti φsur) },
+    exact evariation_on.subsingleton f
+            ((subsingleton_of_subsingleton.image _).anti (t.surj_on_image φ)) },
   let ψ := φ.inv_fun_on t,
-  have ψφs : set.eq_on (φ ∘ ψ) id s := φsur.right_inv_on_inv_fun_on,
-  have ψts : set.maps_to ψ s t := φsur.maps_to_inv_fun_on,
-  have hψ : monotone_on ψ s :=
+  have ψφs : (φ '' t).eq_on (φ ∘ ψ) id := (t.surj_on_image φ).right_inv_on_inv_fun_on,
+  have ψts : (φ '' t).maps_to ψ t := (t.surj_on_image φ).maps_to_inv_fun_on,
+  have hψ : monotone_on ψ (φ '' t) :=
     function.monotone_on_of_right_inv_on_of_maps_to hφ ψφs ψts,
-  change evariation_on (f ∘ id) s ≤ evariation_on (f ∘ φ) t,
-  rw ←eq_of_eq_on (ψφs.comp_left : set.eq_on (f ∘ (φ ∘ ψ)) (f ∘ id) s),
+  change evariation_on (f ∘ id) (φ '' t) ≤ evariation_on (f ∘ φ) t,
+  rw ←eq_of_eq_on (ψφs.comp_left : (φ '' t).eq_on (f ∘ (φ ∘ ψ)) (f ∘ id)),
   exact comp_le_of_monotone_on _ ψ hψ ψts,
 end
 
-lemma comp_eq_of_antitone_on (f : α → E) {s : set α} {t : set β} (φ : β → α)
-  (hφ : antitone_on φ t) (φst : set.maps_to φ t s) (φsur : set.surj_on φ t s) :
-  evariation_on (f ∘ φ) t = evariation_on f s :=
+-- porting note: move to file `data.set.intervals.basic` once the port is over,
+-- and use it in `polynomial_functions_closure_eq_top`
+-- and `topology/continuous_function/weierstrass.lean`
```

Comment:
```suggestion
-- in the file `topology/continuous_function/weierstrass.lean`
```

Reply:
is that to match some kind of regex?

Reply:
No, it's that the first line is the lemma name, and the second line is the file name, and I wanted to make that clear (and I failed :-)

Reply:
My mistake, I should be more careful when reading suggestions.